src/Workspaces/Remote/Core/Services/AssetService.cs

```diff
@@ -107,7 +114,7 @@ private bool EnsureCacheEntryIfExists(Checksum checksum)
             return _assetStorage.TryGetAsset(checksum, out unused);
         }
 
-        private async Task SynchronizeAssetsAsync(ISet<Checksum> checksums, CancellationToken cancellationToken)
+        public async Task SynchronizeAssetsAsync(ISet<Checksum> checksums, CancellationToken cancellationToken)
```

Comment:
Should this be call AddAssetsAsync? The name makes it confusing with your newly added method above which actually calls synchronize assets on ChecksumSynchronizer.